plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java

```diff
@@ -142,17 +142,22 @@ public static boolean isPrestoView(Map<String, String> tableParameters)
 
     public static boolean isHiveOrPrestoView(Table table)
     {
-        return isHiveOrPrestoView(table.getTableType());
+        return isHiveOrPrestoView(table.getTableType(), table.getParameters());
     }
 
-    public static boolean isHiveOrPrestoView(String tableType)
+    public static boolean isHiveOrPrestoView(String tableType, Map<String, String> tableParameters)
     {
-        return tableType.equals(VIRTUAL_VIEW.name());
+        return tableType.equals(VIRTUAL_VIEW.name()) && !isTrinoMaterializedView(tableType, tableParameters);
+    }
+
+    public static boolean isTrinoMaterializedView(Table table)
+    {
+        return isTrinoMaterializedView(table.getTableType(), table.getParameters());
     }
 
     public static boolean isTrinoMaterializedView(String tableType, Map<String, String> tableParameters)
     {
-        return isHiveOrPrestoView(tableType) && isPrestoView(tableParameters) && tableParameters.get(TABLE_COMMENT).equalsIgnoreCase(ICEBERG_MATERIALIZED_VIEW_COMMENT);
+        return tableType.equals(VIRTUAL_VIEW.name()) && isPrestoView(tableParameters) && tableParameters.get(TABLE_COMMENT).equalsIgnoreCase(ICEBERG_MATERIALIZED_VIEW_COMMENT);
```

Comment:
`tableType.equals(VIRTUAL_VIEW.name())` should be redundant, `isPrestoView` should not return true for things that are not a view

`isPrestoView` should imply `isHiveOrPrestoView`. currently it does not only the latter checks table type.

Reply:
Agreed, the existing check was redundant. My logic was I wanted to keep the existing behavior the same. 

I'll remove.